src/common/tse.c

```diff
@@ -661,9 +670,9 @@ tse_task_post_process(tse_task_t *task)
 			 * block.
 			 */
 			/** release lock for CB */
-			D_MUTEX_UNLOCK(&dsp->dsp_lock);
+			D_MUTEX_UNLOCK(&dsp_tmp->dsp_lock);
 			done = tse_task_complete_callback(task_tmp);
-			D_MUTEX_LOCK(&dsp->dsp_lock);
+			D_MUTEX_LOCK(&dsp_tmp->dsp_lock);
```

Comment:
dsp?

Reply:
I think here should take task_tmp's scheduler lock which is dsp_tmp.